### PR TITLE
Add Scala 3 build

### PR DESCRIPTION
**Pull request by ybasket**
Replaces circe-generic-extras with circe-generic by changing the spelling of some fields to snake_case in Scala code.

Based on https://github.com/trace4cats/trace4cats/pull/587

### CODE REVIEW

**Comment by catostrophe** [comment]
Oops, we forgot to edit mergify.yml in previous PRs. Can you add check steps?

**Comment by ybasket** [comment]
> Oops, we forgot to edit mergify.yml in previous PRs. Can you add check steps?

Like 7d83f287?

**Comment by catostrophe** [comment]
You missed one digit pattern.
Copy it from here, please:
https://www.github.com/janstenpickle/hotswap-ref/tree/master/.github/mergify.yml

File: .github/mergify.yml [hunk 1]
```diff
@@ -28,6 +28,8 @@ pull_request_rules:
           - check-success~=Test \(ubuntu-latest, 2\.12\.\d+, adopt@1\.11\)
           - check-success~=Test \(ubuntu-latest, 2\.13\.\d+, adopt@1\.8\)
           - check-success~=Test \(ubuntu-latest, 2\.13\.\d+, adopt@1\.11\)
+          - check-success~=Test \(ubuntu-latest, 3\.\d+, adopt@1\.8\)
+          - check-success~=Test \(ubuntu-latest, 3\.\d+, adopt@1\.11\)
```
Comment (catostrophe) [inline .github/mergify.yml:32]:
```suggestion
          - check-success~=Test \(ubuntu-latest, 3\.\d+\.\d+, adopt@1\.8\)
          - check-success~=Test \(ubuntu-latest, 3\.\d+\.\d+, adopt@1\.11\)
```

File: .github/mergify.yml [hunk 2]
```diff
@@ -45,6 +47,8 @@ pull_request_rules:
...
```
Comment (catostrophe) [inline .github/mergify.yml:51]:
```suggestion
          - check-success~=Test \(ubuntu-latest, 3\.\d+\.\d+, adopt@1\.8\)
          - check-success~=Test \(ubuntu-latest, 3\.\d+\.\d+, adopt@1\.11\)
```

File: project/Dependencies.scala [hunk 3]
```diff
@@ -4,9 +4,10 @@ object Dependencies {
   object Versions {
     val scala212 = "2.12.14"
     val scala213 = "2.13.6"
+    val scala3 = "3.0.1"
 
-    val trace4cats = "0.12.0-RC2"
-    val trace4catsExporterHttp = "0.12.0-RC2"
+    val trace4cats = "0.12.0-RC2+17-d73c7ff3"
```
Comment (ybasket) [inline project/Dependencies.scala:9]:
```suggestion
    val trace4cats = "0.12.0-RC2+18-d1292558"
```

Once https://github.com/trace4cats/trace4cats/pull/591 is merged, fixes the `ArithmeticException` issue in CI.

**Review by catostrophe — APPROVED** [review]

**Pull request merged**